pkg/issuer/acme/http/http.go

```diff
@@ -175,6 +176,7 @@ func testReachability(ctx context.Context, url *url.URL, key string) error {
 		Method: http.MethodGet,
 		URL:    url,
 	}
+	req.Header.Set("User-Agent", pkgutil.CertManagerUserAgent)
```

Comment:
nit: I think header names are meant to be all lowercase

only raising it on the off chance it is causing these test failures 😩

Reply:
wait... Header is a `map[string][]string` - does it need to be initialised somewhere? https://golang.org/src/net/http/header.go 😬

Reply:
Yep it does - and it seems like `NewRequestWithContext` can also handle this for us, so it may be better for us to switch to that and remove `req.WithContext` below too 😄 

/lgtm cancel

Reply:
If we had a unit test for `testReachability` this would have been a lot more obvious of a bug 😢

Reply:
re case: it should be like this as the Go library will use that exact case:
```go
// UserAgent returns the client's User-Agent, if sent in the request.
func (r *Request) UserAgent() string {
	return r.Header.Get("User-Agent")
}
``

Reply:
Added NewRequestWithContext, just noticed we were not using that :cold_sweat: